### ISSUE TITLE
Infer `collect()` / `Collection::make` / `LazyCollection::make` return types for arbitrary inputs (Carbon, DateTime, Jsonable, plain objects)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#808]
## Context

#783 widened `collect()`, `Collection::make()`, and `LazyCollection::make()` stubs to accept `scalar | UnitEnum | null`, closing the false-positive on `collect('some')`. At runtime these entry points accept far more than that: `getArrayableItems()` falls through to `Arr::from()`, which handles `Arrayable`, `Enumerable`, `Traversable`, `WeakMap`, `Jsonable`, `JsonSerializable`, and any `object` (via `(array) $obj`).

As a result, common real-world calls still raise `InvalidArgument`:

```php
collect(now());          // Carbon implements JsonSerializable → [$iso8601String]
collect(new \DateTime);  // plain object → (array) cast → [date=>..., timezone=>..., timezone_type=>...]
collect($anyDto);        // same
```

## What Larastan does

Larastan does not stub `collect()` at all. They use a dynamic return-type extension (`src/ReturnTypes/Helpers/CollectExtension.php` + `src/Support/CollectionHelper.php`) that branches on the input type:

| Input | Return type |
| --- | --- |
| no args | `Collection<int\|string, mixed>` |
| `Enumerable` (Collection, EloquentCollection) | `Collection<TKey, TModel>` |
| `Traversable` / `IteratorAggregate` / `Iterator` | `Collection<TKey, TValue>` from the class's template map |
| empty array | `Collection<int\|string, mixed>` |
| non-empty array | fall through to native PHPDoc |
| anything else (scalar, null, object, Jsonable, JsonSerializable, ...) | `Collection<key-of-toArray, value-of-toArray>` via PHPStan's `$type->toArray()` |

Their assertions (from `tests/Type/data/collection-helper.php`):

```php
collect(1)     // Collection<0, 1>
collect('foo') // Collection<0, 'foo'>
collect(3.14)  // Collection<0, 3.14>
```

Note the literal `0` key (more precise than our current stub's `int`). Larastan also does not cover `Collection::make()` or `LazyCollection::make()`, so those still error on scalars in their plugin.

## Proposed implementation

Mirror Larastan's logic but apply it to all three entry points via Psalm providers, placed in `src/Handlers/Collections/`:

1. `CollectHelperReturnTypeHandler implements FunctionReturnTypeProviderInterface` for `\collect`.
2. `CollectionMakeReturnTypeHandler implements MethodReturnTypeProviderInterface` for `Illuminate\Support\Collection::make` and `Illuminate\Support\LazyCollection::make` (including Eloquent subclasses).

Shared logic in a `Util\CollectionFromValue` helper:

- Input `is TClosure` (only for `LazyCollection::make`), return `static<TKey, TValue>` from the closure's `Generator` return.
- Input implements `Enumerable` / `Arrayable` / `Traversable`: extract `TKey`/`TValue` from the type's template map.
- Input is `TArray`: return its key/value types (native PHPDoc already handles non-empty case, so only fill `Collection<int\|string, mixed>` for empty).
- Input is scalar/UnitEnum/Stringable literal: return `static<0, $value>` (literal `0`, one notch more precise than the merged stub).
- Input is `null`: `static<never, never>`.
- Input implements `Jsonable`: `static<int, string>` if `toJson()` returns a string (ISO8601 from Carbon is the canonical case).
- Input implements `JsonSerializable`: `static<array-key, mixed>` (could narrow further if `jsonSerialize` return type is known).
- Input is plain `object`: `static<string, mixed>` (matches `(array) $obj` property-bag semantics).

Then remove the `make()` stubs and the `scalar | UnitEnum | null` widening of `collect()` introduced in #783 (the provider supersedes them).

## Improvements over Larastan

- **Precise single-element key**: `static<0, $value>` vs Larastan's `int` (actually Larastan already uses `0`, so parity here, not improvement).
- **Static `make()` coverage**: Larastan does not handle `Collection::make($scalar)`; ours would.
- **Jsonable narrowing**: Carbon and Jsonable→string cases could return `Collection<int, string>` concretely.
- **Taint flow**: the security-taint reviewer flagged a pre-existing blind spot: `echo collect(\$_GET['x'])->first()` does not raise a taint warning because the stub's empty body plus the `first()` accessor breaks the flow. A provider can attach `@psalm-flow` behavior directly to the constructed type, closing this gap. Larastan does not model taint at all.

## Acceptance criteria

- All existing `tests/Type/tests/Collection/*.phpt` pass unchanged.
- New PHPT tests cover: `collect(now())`, `collect(new \DateTime)`, `collect(\$somePlainObject)`, `collect(\$jsonSerializable)`, `collect()` no-args, `Collection::make()` no-args, `LazyCollection::make(\$generator)` (bare `Generator` should still raise `InvalidArgument`, per `LazyCollection::__construct`).
- Taint: `echo collect(\$_GET['x'])->first()` triggers `TaintedHtml` (requires collaboration with Collection accessor stubs).
- `Collection::make` stub added in #783 can be removed once the provider covers its cases; `collect()` stub reverts to original narrow PHPDoc (native Laravel-ish) since the provider owns the type.

## References

- Current stubs (post-#783): `stubs/common/Support/Collection.stubphp`, `stubs/common/Support/LazyCollection.stubphp`, `stubs/common/Collections/helpers.stubphp`.
- Laravel runtime: `Arr::from()` at `vendor/laravel/framework/src/Illuminate/Collections/Arr.php:457`, `EnumeratesValues::getArrayableItems()` at `:1090`.
- Larastan source: [`CollectExtension.php`](https://github.com/larastan/larastan/blob/master/src/ReturnTypes/Helpers/CollectExtension.php), [`CollectionHelper.php`](https://github.com/larastan/larastan/blob/master/src/Support/CollectionHelper.php), [`tests/Type/data/collection-helper.php`](https://github.com/larastan/larastan/blob/master/tests/Type/data/collection-helper.php).
